metriken-exposition/src/convert.rs

```diff
@@ -0,0 +1,79 @@
+use std::fs::File;
+use std::io::{BufRead, BufReader, Read, Seek, Write};
+use std::path::Path;
+
+use parquet::errors::ParquetError;
+
+use crate::snapshot::Snapshot;
+use crate::{ParquetOptions, ParquetSchema};
+
+#[derive(Clone, Debug)]
+pub struct MsgpackToParquet {
+    compression_level: i32,
+}
+
+impl Default for MsgpackToParquet {
+    fn default() -> Self {
+        Self {
+            compression_level: 3,
+        }
+    }
+}
+
+impl MsgpackToParquet {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn compression_level(mut self, level: i32) -> Self {
+        self.compression_level = level;
+        self
+    }
+
+    /// Converts a file with metrics in msgpack format to a parquet file.
+    /// Input and putput are file paths.
+    /// If successful, returns the number of rows written out to the parquet
+    /// file.
+    pub fn convert_file_path(
+        self,
+        input: impl AsRef<Path>,
+        output: impl AsRef<Path>,
+    ) -> Result<i64, ParquetError> {
+        self.convert_file_handle(File::open(input)?, File::create(output)?)
+    }
+
+    /// Converts a file with metrics in msgpack format to a parquet file.
+    /// Input and putput are open file handles.
+    /// If successful, returns the number of rows written out to the parquet
+    /// file.
+    pub fn convert_file_handle(
+        self,
+        reader: impl Read + Seek,
+        writer: impl Write + Send,
+    ) -> Result<i64, ParquetError> {
```

Comment:
It's weird to me that this is an `i64`. Is there ever going to be a file with a negative number of rows?

Reply:
Only to reflect whatever the underlying type from the parquet library is: https://github.com/apache/arrow-rs/blob/master/parquet/src/format.rs#L4916

I don't have a preference here -- it isn't clear to me whether returning anything is particularly useful vs just a `Result<(), ParquetError>`.

Reply:
That seems fair to me. If we find that it becomes an issue then we can change it in the future.